op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,407 @@
+}
+
+// NewHandler creates a new flashblocks handler
+func NewHandler(cfg Config, log log.Logger, isLeaderFn func(context.Context) bool) (FlashblockHandler, error) {
+	// Create the hub
+	hub := newHub()
+
+	// Initialize the handler
+	handler := &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+		hub:        hub,
+	}
+
+	// Establish connection to rollup boost if URL is configured
+	if cfg.RollupBoostWsURL == "" {
+		log.Info("rollup boost WebSocket disabled, no URL configured")
+		return handler, nil
+	}
+
+	log.Info("connecting to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
+	conn, _, err := websocket.DefaultDialer.Dial(cfg.RollupBoostWsURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to rollup boost WebSocket: %w", err)
+	}
+	handler.rollupBoostConn = conn
+	log.Info("connected to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
+
+	return handler, nil
+}
+
+// Start initializes the flashblocks handler and starts the WebSocket server and rollup boost listener
+func (h *Handler) Start(ctx context.Context) error {
+	// Start the WebSocket server
+	if err := h.startWebSocketServer(ctx); err != nil {
+		return err
+	}
+
+	// Start the rollup boost listener if connection exists
+	if h.rollupBoostConn != nil {
+		go h.listenToRollupBoost(ctx)
+	}
+
+	return nil
+}
+
+// Stop closes any open WebSocket connections and shuts down the server
+func (h *Handler) Stop() {
+	// Signal the hub to stop if it exists
+	if h.hub != nil {
+		close(h.hub.done)
+	}
+
+	// Close the rollup boost connection if it exists
+	if h.rollupBoostConn != nil {
+		h.log.Info("closing rollup boost WebSocket connection due to context cancellation")
+		h.rollupBoostConn.Close()
+		h.rollupBoostConn = nil
+	}
+
+	// Force close the HTTP server if it's running
+	if h.server != nil {
+		h.log.Info("closing WebSocket server")
+		err := h.server.Close()
+		if err != nil {
+			h.log.Error("Error closing WebSocket server", "err", err)
+		}
+		h.log.Info("WebSocket server closed")
+	}
+}
+
+// BroadcastMessage sends a message to all connected WebSocket clients
+func (h *Handler) BroadcastMessage(message []byte) {
+	h.log.Trace("Broadcasting message to WebSocket clients")
+	h.hub.broadcast <- message
+}
+
+// listenToRollupBoost listens for messages from the rollup boost WebSocket
+func (h *Handler) listenToRollupBoost(ctx context.Context) {
+	defer func() {
+		if h.rollupBoostConn != nil {
+			h.log.Info("closing rollup boost WebSocket connection")
+			h.rollupBoostConn.Close()
+			h.rollupBoostConn = nil
+		}
+	}()
+
+	retryCount := 0
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-h.hub.done:
+			return
+		default:
+			// Check if connection is nil and try to reconnect
+			if h.rollupBoostConn == nil {
+				if retryCount >= MaxReconnectAttempts {
+					h.log.Error("exceeded maximum reconnection attempts to rollup boost WebSocket", "maxRetries", MaxReconnectAttempts)
+					return
+				}
```

Comment:
Let's not return here, in the case that rollup-boost is temporarily down for whatever reason, conductor should always try to reconnect in case it came back (otherwise we have to do a deploy)